blog/content/post/2022-06-17-fobnail_minimal_os.md

```diff
@@ -0,0 +1,253 @@
+
+So we evaluated the feasibility of using microkernel-based OSes. Many
+microkernel-based OSes lack the required drivers, ability to boot another OS,
+and ability to run as DLME payload. Also, since they are designed for an
+embedded environment, they are not portable (across platforms with the same CPU
+architecture), which is a serious limitation that would force us to ship many
+versions of OS for each platform.
+
+As a teaser we include here a table that summarize our research.
+
+| OS      | USB host driver  | USB EEM driver   | Network stack     | TPM driver        | OS portability | Bootloader capabilities | C library    | Microkernel | CPU Architecture support | Bootable by SKL | License | Score |
+| ------- | ---------------- | ---------------- | ----------------- | ----------------- | -------------- | ----------------------- | -------------| ----------- | ------------------------ | --------------- | ------- | ----- |
+| Zephyr  | Yes (+2)         | Yes (+2)         | Yes (+2)          | PoC available (0) | Limited (-1)   | No (0)                  | Yes (+2)     | No (0)      | Good (+1) [^4]           | No (0) [^6]     |  OK (0)  | 8     |
+| Xous    | No (0)           | No (0)           | Yes (+2)          | No (0)            | Limited (-1)   | No (0)                  | No (0)       | Yes (+1)    | RISC-V only (-1)         | No (0)          |  OK (0)  | 1     |
+| seL4    | No (0) [^1]      | No (0) [^2]      | Yes (+2)          | No (0)            | Limited (-1)   | No (0)                  | Yes (+2)     | Yes (+1)    | Good (+1) [^3]           | Yes (+2)        | OK (but problematic with Genode) (0)  | 7     |
+| Linux   | Yes (+2)         | Yes (+2)         | Yes (+2)          | Yes (+2)          | Yes (+1)       | Yes (kexec) (+2)        | Yes (+2)     | No (0)      | Good (+1) [^5]           | Yes (+2)        |  OK (0)  | 16    |
+| LK      | No (0)           | No (0)           | Limited (-2) [^7] | No (0)            | Yes (+1)       | No (0)                  | Limited (-2) | No (0)      | Good (+1) [^8]           | No (0)          |  OK (0)  | -2    |
+| Fuchsia | Limited (0) [^9] | No (0)           | Yes (+2)          | Limited (0) [^10] | Yes (+1)       | Yes (mexec) (+2)        | Yes (+2)     | Yes (+1)    | Good (+1) [^11]          | No (0)          |  OK (0)  | 7     |
+
+As we can see, multiple OSes were taken into account and a lot of requirements.
+If you are interested in the meanings of the numbers here, please check the
+full [report](https://fobnail.3mdeb.com/minimal-os-for-fobnail/) available on
+Fobnail Project official website.
+
+## Why Linux?
+
+We have decided to use Linux for building minimal OS because it already has
+everything we need, and other OSes would require a significant amount of work.
+In the future, we may use another OS.
+
+## Reference minimal OS for Fobnail Project
+
+We build a minimal OS image for Fobnail Project by using [Yocto
+Project](https://www.yoctoproject.org/) our
```

Comment:
```suggestion
Project](https://www.yoctoproject.org/), our
```

Reply:
fixed https://github.com/3mdeb/news-and-ideas/pull/496/commits/acea6f8ce69e7cd420e09461d14ba223d9597e59#diff-5f2030137d33048b104fbebb6f1f3ce21c3e5522e57c8841c0fc8be5ceb88e2cL78